backend/uchicago/src/coursesearch_parser.py

```diff
@@ -0,0 +1,41 @@
+import collections
+import re
+import scrapydo
+
+from .course import Course
+from .section import Section
+
+from .coursesearch_spider import CourseSearchSpider
+
+class CourseSearch(object):
+  init = False
+  def __init__(self, label, id):
+    if not CourseSearch.init:
+      scrapydo.setup()
+      CourseSearch.init = True
+    self.label = label
+    self.id = id
+    
+  @property
+  def courses(self):
+  
+    settings = {
+        'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)',
+        'DOWNLOADER_MIDDLEWARES': {
+            'src.coursesearch_spider.StateManagerMiddleware': 543,
+        }
+    }
+    
+    spider = CourseSearchSpider
+    ret = scrapydo.run_spider(spider, settings=settings, label = self.label, id = self.id)
```

Comment:
I think we'll still want to have the sharding though. There could be scheduled classes that get clipped after the 250 cap without day of week sharding otherwise. Perhaps it would be best to do eight passes? Once without sharding, then once per day of week. To save on resources and not get UChicago too upset, we could shard if we observe that the unsharded scrape exceeds 250.